lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -525,47 +520,49 @@ def mock_run(args, **kwargs):
                 args, returncode=0, stdout="Successfully Unpacked!", stderr=None
             )
 
+        def mock_resolve(_path, _strict=False):
+            """In this scenario, there are no symlinks,
+            so resolve() should never be called."""
+            raise AssertionError("Unexpected call to Path.resolve()")
```

Comment:
Not at all sure how I feel about that, but I'll let it go. 😆

Reply:
Yeah...I have all sorts of conflicting desires on this one.

I would like to have 100% coverage of the CUT, with 100% execution of the unit test code.  This suggests that I should just remove the unused mock.

On the other hand, I know that the CUT _would_ call `resolve()` if execution happened to follow the other branch of a conditional in the function, so there is a real danger of calling it, which suggests that we should have a rigorous mock.

On the gripping hand, I'm not sure how much it would really matter if the real `resolve()` got called, and the results of that might be just as obvious as raising the assertion exception.

I don't really want the extra complication in the unit test, especially when the key part of the code won't even execute because we never call the mock....

🦐